domain/src/main/java/org/oppia/android/domain/oppialogger/analytics/ApplicationLifecycleObserver.kt

```diff
@@ -71,4 +97,58 @@ class ApplicationLifecycleObserver @Inject constructor(
       }
     }
   }
+
+  private fun logRelativeCpuUsageInBackground(
+    timeIntervalInMillis: Long,
+    currentApplicationState: ApplicationState,
+    currentScreen: ScreenName
+  ) {
+    CoroutineScope(backgroundDispatcher).launch {
+      while (true) {
+        val previousCpuUsageParameters = performanceMetricsController.getLastCpuUsageParameters()
+        val currentCpuUsageParameters = performanceMetricsController.getCurrentCpuUsageParameters(
+          currentApplicationState
+        )
+        val relativeCpuUsage = performanceMetricsController.getRelativeCpuUsage(
+          previousCpuUsageParameters, currentCpuUsageParameters
+        )
+        val applicationState = calculateRelativeApplicationState(
+          previousCpuUsageParameters.applicationState,
+          currentApplicationState
+        )
+        performanceMetricsLogger.logCpuUsage(currentScreen, applicationState, relativeCpuUsage)
+        performanceMetricsController.cacheCpuUsageParameters(currentCpuUsageParameters)
+        delay(timeIntervalInMillis)
+      }
+    }
+  }
+
+  private fun calculateRelativeApplicationState(
+    previousApplicationState: ApplicationState,
+    currentApplicationState: ApplicationState
+  ): ApplicationState {
+    return if (
+      previousApplicationState == ApplicationState.APP_IN_FOREGROUND &&
+      currentApplicationState == ApplicationState.APP_IN_BACKGROUND
```

Comment:
FWIW this can probably be done much more compactly using nested ``when`` statements, and possibly more robustly.

Reply:
done.